image-toolkit/sbin/startup.sh

```diff
@@ -75,21 +110,47 @@ webcam_stream_loop() {
             }" &
         CVLC_PID=$!
 
-        # Monitor the video device using udevadm monitor
-        # If device is unplugged, kill existing clvc instance to release /dev/video0
-        udevadm monitor --udev -s video4linux | while read -r line; do
-            if [[ "$line" == *"remove"*"/video4linux/video$VIDEO_DEVICE_NO"* ]] ; then
-                echo "Device unplugged, proceed to kill cvlc"
-                kill $CVLC_PID 2>/dev/null || :
-                break
+        echo "Waiting for cvlc or udevadm to exit"
+        wait -fn -p TERMINATED_PID $UDEVADM_PID $CVLC_PID
+        if [[ $TERMINATED_PID -eq $CVLC_PID ]]; then
+            echo "cvlc exited"
+        else
+            echo "udevadm exited"
+        fi
+
+        echo "Sending SIGTERM to cvlc and udevadm"
+        kill $UDEVADM_PID $CVLC_PID
+
+        echo "Waiting for cvlc and udevadm to exit"
+        # timeout returns 124 if the command times out
+        timeout 3s wait -f $UDEVADM_PID $CVLC_PID; if [ $? -eq 124 ]; then
+            echo "Timeout waiting for cvlc and udevadm to exit, sending SIGKILL"
+            kill -9 $UDEVADM_PID $CVLC_PID
+        fi
```

Comment:
wait is a shell command and does not work with timeout.

Reply:
Thx. Made the change